up42/http/client.py

```diff
@@ -0,0 +1,42 @@
+import functools
+from typing import Dict, List, Optional
+
+from up42.http import config, oauth
+
+
+class Client:
+    def __init__(self, auth: oauth.Up42Auth):
+        self.auth = auth
+
+    @property
+    def token(self):
+        return self.auth.token.access_token
+
+
+def _merge(left: Optional[config.CredentialsSettings], right: Optional[config.CredentialsSettings]):
+    if all([left, right]):
+        raise MultipleCredentialsSources("Multiple sources of credentials provided")
+    return left or right
+
+
+def create(
+    credential_sources: List[Dict],
```

Comment:
```suggestion
    credential_sources: list[dict],
```
since we won't be supporting python 3.8 anymore, we can just use the default dict and list objects natively.

Reply:
same for other occurrences in this PR.

Reply:
I'd like to address that in a follow up PR to include all benefits that 3.9 has over 3.8 in all source code